test/hack/checkLabels/main.go

```diff
@@ -147,48 +167,66 @@ func getMissingLabels() ([]string, []string, error) {
 
 	// Get the list of labels with prefix "cancel_if_"
 	cancelLabels := getLabels(rulesList, "cancel_if_")
-	targetLabels, sourceLabels := getTargets(alertConf)
+	targetLabels, _ := getTargetsAndSources(alertConf, "")
 
 	for _, cancelLabel := range cancelLabels {
 		var i = 0
 
 		for _, targetLabel := range targetLabels {
 			if cancelLabel == targetLabel {
 				i++
+
+				_, source := getTargetsAndSources(alertConf, cancelLabel)
```

Comment:
I think there is a flaw here, were we check for a single source matcher. I agree this is what we intend to test here, so keeping this is ok, but we should comment that this is a limitation.
Something like:
```
// Limitation: we check the first source matchers found as we expect only source matcher for each target matcher. But there could be multiple and this part could be improved.
```

Reply:
In the comment, I'd say "we expect at least one" instead of "only one".

Reply:
Yeah good point, I changed that in my suggestion

Reply:
I added the comment.